actors/market/src/lib.rs

```diff
@@ -534,6 +538,9 @@ impl Actor {
     /// extra info about verified deals.
     /// Sectors' deals are activated in parameter-defined order.
     /// Each sector's deals are activated or fail as a group, but independently of other sectors.
+    /// Note that confirming all deals fit within a sector is the caller's responsibility
+    /// (and is implied by confirming the sector's data commitment is derived from the deal peices).
+    // see https://github.com/filecoin-project/builtin-actors/issues/1308
     fn batch_activate_deals(
```

Comment:
I've refactored this to extract as much shared code as practical (in preactivate_deal). The methods follow very similar patterns, but batch_activate_deals is defined to succeed/fail on a per-sector basis, while sector_content_changed is just per-deal.